### PR TITLE
Modify system-recovery

### PR DESCRIPTION
**Pull request by sh95jeon**
### Change Description ###

- Action of system-recovery is changed, so change boot process.
- There is no more config file with system-recovery, so removed configuration related section.

### Bugs Fixed ###

- No related bug

### API Changes ###

- NON ACR

### CODE REVIEW

**Comment by mijin85cho** [comment]
Hello, @annie-samsung Would you please assign one SRI-B member to review this PR?
As most of the update is removing the out-of-dated content, I think reviewing updated sentence would not be a big load.

One concern - It is recommended not to use "will" in dev doc.
**Reference :** https://developers.google.com/style/tense

**Comment by annie-samsung** [comment]
> Hello, @annie-samsung Would you please assign one SRI-B member to review this PR?
> As most of the update is removing the out-of-dated content, I think reviewing updated sentence would not be a big load here.
> One concern - It is recommended not to use "will" in dev doc.
> Reference : https://developers.google.com/style/tense

@mijin85cho, sure I shall take up the review.

File: docs/platform/porting/recovery.md [hunk 1]
```diff
@@ -36,492 +36,10 @@ the `/system` directory, and other filesystems (such as `/opt` and
...
-        frames_num = 0;
-        current_frame = 0;
-        img_type = "no-alpha"; // alt: "no-alpha";
-    };
-};
-```
+4. The `system-recovery` program search `tizen-recovery.img` from removable storage.
+`tizen-recovery.img` has compressed partition image and config file.
+`tizen-recovery.img` is found from removable storage, `system-recovery` program decompress `tizen-recovery.img`
+and dump partition image to appropicate block with config file.
```
Comment (annie-samsung) [inline docs/platform/porting/recovery.md:42]: Suggest to rephrase, please check if the below statement is technically accurate.

The `system-recovery` program searches the removable storage for `tizen-recovery.img` which has the  compressed partition image and config file. The `system-recovery` program decompress `tizen-recovery.img`, and dumps the partition image to the appropriate block with config file.


Have a question here.. does the config file have the block information, or the config file is copied to the block along with the partition image?

File: docs/platform/porting/recovery.md [hunk 2]
```diff
@@ -36,492 +36,10 @@ the `/system` directory, and other filesystems (such as `/opt` and
 `/opt/usr`) below the `/system` directory. The script starts a shell on the serial console and launches the
 `system-recovery` program.
 
-4. The `system-recovery` program can work as either an interactive GUI or
-a non-interactive dispatcher, executing 1 (non-interactive) or
-more (interactive) actions. The actions are configured in the
-`/usr/share/system-recovery/system-recovery.cfg` file.
-
-## Configuration File
```
Comment (annie-samsung) [inline docs/platform/porting/recovery.md:44]: Are lines 44 to 524 also removed in the new commit?

**Review by annie-samsung — COMMENTED** [review]
Please find my comments

File: docs/platform/porting/recovery.md (hunk 2 above)
Comment (sh95jeon) [reply docs/platform/porting/recovery.md:44]: Yes, that section is describing old configuration which is not used current.

File: docs/platform/porting/recovery.md (hunk 1 above)
Comment (sh95jeon) [reply docs/platform/porting/recovery.md:42]: > Suggest to rephrase, please check if the below statement is technically accurate.

Thank you for suggestion. I looks good, i will fix it 👍 

> Have a question here.. does the config file have the block information, or the config file is copied to the block along with the partition image?

Config file have block information. `system-recovery` program just use it for find proper block.

Comment (annie-samsung) [reply docs/platform/porting/recovery.md]: @sh95jeon  
Thank you for clarifying regarding the config file, can the second line be modified like this:

The `system-recovery` program decompress `tizen-recovery.img`, and dumps the partition image to the appropriate block as mentioned in the config file.

Comment (sh95jeon) [reply docs/platform/porting/recovery.md]: @annie-samsung Thank you for reviewing 😄 I fixed it :)

**Review by annie-samsung — APPROVED** [review]
Looks good to me, approving the changes.

**Review by mijin85cho — APPROVED** [review]
Thanks for the updates and review.

**Pull request merged**